cql3/statements/select_statement.cc

```diff
@@ -404,26 +405,52 @@ select_statement::do_execute(query_processor& qp,
 
     auto key_ranges = _restrictions->get_partition_key_ranges(options);
 
+    bool send_tablet = false;
+
+    auto token = dht::token();
+    unsigned shard;
+
+    if (key_ranges.size() > 0 && query::is_single_partition(key_ranges.front())) {
```

Comment:
Shouldn't this be key_ranges.size() == 1? If the driver uses tablet-awareness only for single-key queries, then we will keep sending tablet info all the time needlessly for such queries, wasting CPU and bandwidth.